tools/mmp/resolver.cs

```diff
@@ -141,25 +142,31 @@ public AssemblyDefinition Resolve (AssemblyNameReference reference, ReaderParame
 			if (assembly != null)
 				return assembly;
 
+			foreach (var directory in ExtraSearchDirectories) {
+				assembly = SearchDirectory (name, directory, true);
+				if (assembly != null)
+					return assembly;
+			}
+
 			return null;
 		}
 
-		AssemblyDefinition SearchDirectory (string name, string directory)
+		AssemblyDefinition SearchDirectory (string name, string directory, bool recursive = false)
```

Comment:
That behaviour is not documented with the option. It's also not a behaviour we have elsewhere.

TBH I'm not sure it's one we should have... pointing to a project directory means you can get debug or release (and other different) configuration *randomly* which would be painful to identify/debug

Reply:
We need to specify a lot of search directories for our use-case, having recursive probing would help us not have to specify around 50 directories for searching. If we pass in all the assemblies, only the root assembly's directory is probed, so if you have a layout like:
```
build
- Addins
-- VersionControl
--- VersionControl.dll (depends on Ide.dll)
--- VersionControl.Git.dll (depends on VersionControl.dll)
- bin
-- Core.dll
-- Ide.dll
```

VersionControl.Git cannot be loaded unless the VersionControl directory is specified too. Having to manually provide all search paths is doable, but probably tedious. I can revert this change if we need to.

Reply:
I think we should either:

- Document that --extra-directories are searched recursively (right now it isn't)
- Make it an option and default to no.

I'm not sure which I prefer.

In general, recursive directly searching is scary, as @spouliot mentioned with Release/Debug.

Reply:
The potential Debug/Release randomness can be mitigated by showing an error if multiple files are found (which should be easy enough to implement in `DirectoryGetFile` below).

I also think the code would be more self-documenting if the variable was named something like `RecursiveSearchDirectories` instead of `ExtraSearchDIrectories`.